### PR TITLE
ESLint

### PR DESCRIPTION
**Pull request by ChristianMurphy**
Replaces Standard with ESLint.
ESLint is set to use Standard ESLint configuration.
Then adds additional validations.

ref https://github.com/feross/eslint-config-standard

### CODE REVIEW

File: lib/usage.js [hunk 1]
```diff
@@ -158,11 +160,11 @@ module.exports = function (yargs) {
...
-        if (~options.boolean.indexOf(key)) type = '[boolean]'
-        if (~options.count.indexOf(key)) type = '[count]'
-        if (~options.string.indexOf(key)) type = '[string]'
-        if (~options.normalize.indexOf(key)) type = '[string]'
-        if (~options.array.indexOf(key)) type = '[array]'
+        if (!options.boolean.indexOf(key)) type = '[boolean]'
+        if (!options.count.indexOf(key)) type = '[count]'
+        if (!options.string.indexOf(key)) type = '[string]'
+        if (!options.normalize.indexOf(key)) type = '[string]'
+        if (!options.array.indexOf(key)) type = '[array]'
```
Comment (isaacs) [inline lib/usage.js]: Wrong.

File: lib/usage.js [hunk 2]
```diff
@@ -247,11 +249,11 @@ module.exports = function (yargs) {
...
-        if (~options.boolean.indexOf(alias)) yargs.boolean(key)
-        if (~options.count.indexOf(alias)) yargs.count(key)
-        if (~options.string.indexOf(alias)) yargs.string(key)
-        if (~options.normalize.indexOf(alias)) yargs.normalize(key)
-        if (~options.array.indexOf(alias)) yargs.array(key)
+        if (!options.boolean.indexOf(alias)) yargs.boolean(key)
+        if (!options.count.indexOf(alias)) yargs.count(key)
+        if (!options.string.indexOf(alias)) yargs.string(key)
+        if (!options.normalize.indexOf(alias)) yargs.normalize(key)
+        if (!options.array.indexOf(alias)) yargs.array(key)
```
Comment (isaacs) [inline lib/usage.js]: Wrong.  Bitwise NOT !== boolean NOT.  `~` is only falsey if the value is `-1` (ie, the thing is not found).  `!` is falsey if the value is `-1` or > `0` (ie, if the thing is either not found, or found after the first position.)

File: lib/usage.js (hunk 1 above)
Comment (ChristianMurphy) [inline lib/usage.js]: restored original bitwise not

File: lib/usage.js (hunk 2 above)
Comment (ChristianMurphy) [inline lib/usage.js]: restored original bitwise not

**Comment by bcoe** [comment]
Hey @ChristianMurphy I really appreciate the hard work you put into this pull request, but a big part of why I love @feross' standard is that it leaves no `. eslintrc` to argue about.

Rather than accepting this pull request, I encourage you to start a conversation regarding some of these stylistic additions on the `standard` repo, I will happily update yargs' style to match any additions to `standard`.

**Comment by ChristianMurphy** [comment]
Thanks @bcoe and @isaacs for the prompt feedback. :smile: 

@bcoe would you be interested in any of the changes made here being merged, but not being enforced?

The most notable changes being
- strict mode
- replacing `var === undefined` with `typeof var === 'underfined'`
- replacing `throw Error()` with `throw new Error()`
- applying standard style on examples

**Pull request closed without merging**